src/player.h

```diff
@@ -1182,6 +1239,9 @@ class Player final : public Creature, public Cylinder
 		std::map<uint32_t, int32_t> storageMap;
 
 		std::vector<OutfitEntry> outfits;
+		std::vector<ProtocolSpectator*> spectators;
```

Comment:
I don't think it's safe to store raw pointers to objects managed by shared_ptrs (unless you can guarantee their lifetimes).